weave/trait/pairing/basic_open_close_installation_state_trait.proto

```diff
@@ -0,0 +1,116 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file specifies the basic open close installation state trait.
+ */
+
+syntax = "proto3";
+
+package weave.trait.pairing;
+
+import "wdl/wdl_options.proto";
+
+option java_outer_classname = "WeaveInternalBasicOpenCloseInstallationStateTrait";
+option objc_class_prefix = "SCM";
+
+/**
+ *  @brief
+ *    Basic Open Close Installation State Trait
+ *
+ *  A Weave Common service published trait that provides a means for an
+ *  application to set a an open close sensor device to different modes
+ *  during installation, and to query the current installation mode of
+ *  the device.
+ *
+ *  The consumer of this information is any actor driving pairing flows
+ *  for open close sensor devices.
+ */
+message BasicOpenCloseInstallationStateTrait {
+  option (wdl.message_type) = TRAIT;
+
+  option (wdl.trait) = {
+    stability: ALPHA,
+    vendor_id: 0x0000,
+    id:        0x1402,
+    version:   1
+  };
+
+  //  @brief
+  //    Modes during installation
+  //
+  //  Different user interface modes used during device installation
```

Comment:
Omit the notion of UI - given that, I think the comment might be redundant.

Reply:
Done